lib/cocoapods-core/specification/consumer.rb

```diff
@@ -110,6 +110,12 @@ def user_target_xcconfig
         merge_values(attr, value_for_attribute(:xcconfig), value_for_attribute(:user_target_xcconfig))
       end
 
+      # @return [Array<String>] A hash where the key represents the
+      #         paths of the on_demand_resources to copy and the values the paths of
+      #         the on_demand_resources that should be copied.
+      #
+      spec_attr_accessor :on_demand_resources
```

Comment:
I think we would want this to be a `Hash` that allows Pod authors to specify the tag (see my comment [here](https://github.com/CocoaPods/Xcodeproj/pull/742#discussion_r415153967))

example:
```ruby
s.on_demand_resources = {
    'Tag1' => ['file1.png', 'file2.png']
}
```

we also may need to provide a way for Pod authors to specify the category of the tag. The category determines how the resource gets fetched. To quote the [docs](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/On_Demand_Resources_Guide/Tagging.html#//apple_ref/doc/uid/TP40015083-CH3-SW1):

> **Initial install tags**. The resources are downloaded at the same time as the app. The size of the resources is included in the total size for the app in the App Store. The tags can be purged when they are not being accessed by at least one NSBundleResourceRequest object.
>
>**Prefetch tag order**. The resources start downloading after the app is installed. The tags will be downloaded in the order in which they are listed in the Prefetched tag order group.
>
> **Dowloaded only on demand**. The tags are downloaded when requested by the app.

Reply:
good idea. I will refer to your design for implementation.

Reply:
implementation https://github.com/CocoaPods/Core/pull/622/commits/f240511fef0d7c56128a336aee7c00f91b718658